### PR TITLE
[CustomHandler]Fix options parsing

### PR DESCRIPTION
**Pull request by pragnagopa**
PR fixes parsing customHandler options to support using a binary:

```json
"customHandler": {
		"enableForwardingHttpRequest": true,
		"description": {
			"defaultExecutablePath": "GoCustomHandlers.exe",
			"arguments": ["-hello"]
		}
	}
```

and using a a runtime that is part of the system path such as

```json
"customHandler": {
		"description": {
			"defaultExecutablePath": "node",
			"arguments": ["server.js"]
		}
	}
```

### CODE REVIEW

**Comment by pragnagopa** [comment]
@anthonychu - FYI.

**Comment by anthonychu** [comment]
Cool. Thanks! What's the current behavior? Does it require using the full path (`/home/site/wwwroot/myapp`) if it's a binary supplied in the app?

**Comment by pragnagopa** [comment]
>  Does it require using the full path (/home/site/wwwroot/myapp) if it's a binary supplied in the app?

Binary without `arguments` works. Yes, you need to specify full path if using a binary and arguments

File: src/WebJobs.Script/Workers/Http/HttpWorkerDescription.cs [hunk 1]
```diff
@@ -42,19 +42,18 @@ public override void ApplyDefaultsAndValidate(string inputWorkerDirectory, ILogg
 
             ExpandEnvironmentVariables();
 
-            // If DefaultWorkerPath is not set then compute full path for DefaultExecutablePath from WorkingDirectory.
+            // If DefaultWorkerPath is not set then compute full path for DefaultExecutablePath from WorkingDirectory and check if DefaultExecutablePath exists
             // Empty DefaultWorkerPath or empty Arguments indicates DefaultExecutablePath is either a runtime on the system path or a file relative to WorkingDirectory.
             // No need to find full path for DefaultWorkerPath as WorkerDirectory will be set when launching the worker process.
             // DefaultWorkerPath can be specified as part of the arguments list
-            if (string.IsNullOrEmpty(DefaultWorkerPath) && !string.IsNullOrEmpty(DefaultExecutablePath) && !Path.IsPathRooted(DefaultExecutablePath) && !Arguments.Any())
+            if (!string.IsNullOrEmpty(DefaultExecutablePath) && !Path.IsPathRooted(DefaultExecutablePath))
```
Comment (yojagad) [inline src/WebJobs.Script/Workers/Http/HttpWorkerDescription.cs:49]: >if (!string.IsNullOrEmpty(DefaultExecutablePath) && !Path.IsPathRooted(DefaultExecutablePath)) [](start = 12, length = 94)

Should this have been if (string.IsNullOrEmpty(DefaultExecutablePath) && !Path.IsPathRooted(DefaultExecutablePath)) based on the description above ?

Comment (pragnagopa) [reply src/WebJobs.Script/Workers/Http/HttpWorkerDescription.cs:49]: DefaultExecutablePath is required. this logic is for figuring whether to append worker directory or not

Comment (yojagad) [reply src/WebJobs.Script/Workers/Http/HttpWorkerDescription.cs:49]: Ah, okay. Make sense

**Review by yojagad — APPROVED** [review]

**Pull request merged**